service/service.go

```diff
@@ -19,13 +19,20 @@ type AgentService struct {
 	agent                 *agent.LogAgent
 	persister             operator.Persister
 	persisterShutdownFunc persist.PersisterShutdownFunc
+	pprofProfiler         *pprofProfiler
 }
 
 // Start will start the stanza agent.
 func (a *AgentService) Start(s service.Service) error {
 	a.agent.Info("Starting stanza agent")
 	if err := a.agent.Start(a.persister); err != nil {
-		a.agent.Errorw("Failed to start stanza agent", zap.Any("error", err))
+		a.agent.Errorw("Failed to start stanza agent", zap.Error(err))
+		a.cancel()
+		return nil
+	}
+
+	if err := a.pprofProfiler.Start(); err != nil {
```

Comment:
Might be worth a comment here that profiling will actually only start if enabled. I had to scan back through to double check that was the case.

Reply:
Good call, added a comment here and above the start function itself.